tests/integration/utils.py

```diff
@@ -0,0 +1,187 @@
+
+
+# Get resource schema from OpenAPI specification
+def get_resource_schema(self, resource):
+    return self.openapi['definitions'][resource]['properties']
+
+
+# Helper function to make a web request and lightly validate the response
+def make_request(self, endpoint, expected_status_code,
+                 params=None,
+                 max_elapsed_seconds=5):
+    """
+    Keyword arguments:
+    * endpoint -- the endpoint to request
+    * expected_status_code -- expected HTTP status code
+    * params -- [optional] key-value pairs parameters (default: None)
+    * max_elapsed_seconds -- [optional] maximum elapsed times (default: 5)
+    Return:
+    A response object contains a server’s response to an HTTP request
+    """
+
+    requested_url = f'{self.base_url}{endpoint}'
+    response = self.session.get(requested_url, params=params)
+    logging.debug(f'Sent request to {requested_url}')
+    # Response status code should be as expected
+    status_code = response.status_code
+    self.assertEqual(status_code, expected_status_code)
+
+    # Response time should less then max_elapsed_seconds
+    elapsed_seconds = response.elapsed.total_seconds()
+    logging.debug(f'Request took {elapsed_seconds} second(s)')
+    self.assertLess(elapsed_seconds, max_elapsed_seconds)
+
+    return response
+
+
+# Check the schema of response match OpenAPI specification
+def check_schema(self, response, schema):
+    # Helper function to get attributes of the schema
+    def __get_schema_attributes():
+        return schema['attributes']['properties']
+
+    # Helper function to map between OpenAPI data types and python data types
+    def __get_attribute_type(attribute):
+        types_dict = {
+            'string': str,
+            'integer': int,
+            'int32': int,
+            'int64': int,
+            'float': float,
+            'double': float,
+            'boolean': bool,
+            'array': list,
+            'object': dict
+        }
+        if '$ref' in attribute:
+            openapi_type = __get_object_type(attribute['$ref'])
+        elif 'properties' in attribute:
+            return dict
+        elif 'format' in attribute and attribute['format'] in types_dict:
+            openapi_type = attribute['format']
+        elif 'type' in attribute:
+            openapi_type = attribute['type']
+        else:
+            logging.warning('OpenAPI property contains no type or properties')
+            return None
+
+        return types_dict[openapi_type]
+
+    # Helper function to get type of referenced object
+    def __get_object_type(object_path):
+        keys = object_path.split("#/")[-1].split("/")
+        schema = self.openapi
+        for key in keys:
+            schema = schema[key]
+        return schema['type']
+
+    # Helper function to check resource object schema
+    def __check_resource_schema(resource):
+        # Check resource type
+        self.assertEqual(resource['type'], schema['type']['example'])
```

Comment:
The skeleton was recently updated here https://github.com/osu-mist/express-api-skeleton/blob/master/tests/integration/utils.py#L140 so that it uses `enum[0]` instead of `example`. This is because it was decided that our swagger documents should use a singleton `enum` for the value of the top level `type` key in the jsonapi response. Could you change this and also update `swagger.yaml` so that it follows this convention?